PR:Add Product entity and integrate with ApplicationDbContext

Ecomont opened the pull request:
This pull request introduces a new `Product` entity to the application and updates the database context to support it. The main changes are the addition of the `Product` model and its integration into Entity Framework for data access.

Ecomont committed:
Add Product entity and integrate with ApplicationDbContext

Ecomont committed:
Add PurchaseProduct entity and integrate with DbContext

alejandrogomezl commented on src/AppForSEII2526.API/Models/Product.cs:
```diff
@@ -0,0 +1,18 @@
+﻿namespace AppForSEII2526.API.Models
+{
+    public class Product
```
Please, check Class Diagram attributes 

<img width="302" height="364" alt="image" src="https://github.com/user-attachments/assets/093f0f9c-f294-4a10-88ca-a11c9bef1725" />

Ecomont committed:
Refactor Product model

Ecomont committed:
Add delivery and payment system entities and relationships

Updated `ApplicationDbContext` to include `DbSet` properties for new entities: `PurchaseProduct`, `PurchaseDelivery`, `PaymentMethod`, `CreditCard`, `PayPal`, `Bizum`, and `DeliveryDriver`. Added `OnModelCreating` override for model configuration.

Introduced `PaymentMethod` as an abstract base class and added derived classes: `Bizum`, `CreditCard`, and `PayPal`. Added `DeliveryDriver` class to represent delivery personnel.

Created `PurchaseDelivery` class to manage delivery assignments with priorities, including a `PriorityType` enum. Updated `DeliveryAssignment` and `PurchaseOrder` classes to integrate these new features.

Added a class diagram (`UC - Schedule delivery.cd`) to visually represent the relationships between the new and updated classes.

Ecomont committed:
Update dbcontext

Ecomont committed:
Add Brand entity and DbSet properties to ApplicationDbContext

Ecomont committed:
Refactor property initializations and fix syntax issues

Ecomont committed:
Fix

alejandrogomezl reviewed (approved):
Well done!!
